tracer/src/Datadog.Trace/Logging/DirectSubmission/Formatting/LogFormatter.cs

```diff
@@ -42,12 +45,32 @@ internal class LogFormatter
             _service = string.IsNullOrEmpty(serviceName) ? null : serviceName;
             _host = string.IsNullOrEmpty(settings.Host) ? null : settings.Host;
             _globalTags = string.IsNullOrEmpty(settings.GlobalTags) ? null : settings.GlobalTags;
+            _enrichedGlobalTags = new Lazy<string?>(EnrichGlobalTagsStringWithSourceLinkData);
             _env = string.IsNullOrEmpty(env) ? null : env;
             _version = string.IsNullOrEmpty(version) ? null : version;
         }
 
         internal delegate LogPropertyRenderingDetails FormatDelegate<T>(JsonTextWriter writer, in T state);
 
+        private string? EnrichGlobalTagsStringWithSourceLinkData()
+        {
+            if (_globalTags?.Contains(CommonTags.GitCommit) == true &&
+                _globalTags?.Contains(CommonTags.GitRepository) == true)
+            {
+                // The global tags already contain git data
+                return _globalTags;
+            }
+
+            var tags = SourceLinkTagsProvider.Instance.GetGitTagsFromSourceLink();
+            if (tags == null)
+            {
+                return _globalTags;
+            }
+
+            var formattedGitTags = string.Join(",", tags.Select(kvp => $"{kvp.Key}:{kvp.Value}"));
+            return _globalTags == null ? formattedGitTags : $"{_globalTags},{formattedGitTags}";
+        }
```

Comment:
Great feedback, thanks a lot! I reworked things https://github.com/DataDog/dd-trace-dotnet/commit/52d6dff455366205f331e1c766c27cb86804cc07

> If the tags contain one of commit or repo, we could get duplicate/conflicting commit/repo data. How should we handle that?
I believe that can't happen because I'm explicitly checking that they are both available. Source Code Integration requires **both** the `git.commit.sha` and the `repository_url` to be tagged.